typescript/tuple_tree.ts

```diff
@@ -39,6 +39,22 @@ export function safeParseInt(str?: string): number | undefined {
     }
 }
 
+/**
+ * Parses a bigint safely, returning undefined if the number cannot be represented as a decimal
+ * string
+ */
+export function safeParseBigInt(str?: string): bigint | undefined {
+    if (str !== undefined) {
+        try {
+            return BigInt(str);
+        } catch (err) {
```

Comment:
Can we limit what we catch here?

Reply:
`BigInt` will throw `TypeError`, `RangeError` and `SyntaxError` which are 3 out of the 8 errors that Javascript natively supports. The other 5, in this context, cannot happen.

Reply:
Removed `safeParseBigInt`